src/app/qgisapp.cpp

```diff
@@ -9499,6 +9503,46 @@ QString QgisApp::saveAsVectorFileGeneral( QgsVectorLayer *vlayer, bool symbology
   return vectorFilename;
 }
 
+QString QgisApp::saveAsPointCloudLayer( QgsPointCloudLayer *pclayer )
+{
+  QgsPointCloudLayerSaveAsDialog *dialog = new QgsPointCloudLayerSaveAsDialog( pclayer );
+
+  dialog->setMapCanvas( mMapCanvas );
+
+  dialog->setAddToCanvas( true );
+
+  QString vectorFilename;
+  if ( dialog->exec() == QDialog::Accepted )
+  {
+    QgsPointCloudLayerExporter exp( pclayer );
+
+    QgsCoordinateReferenceSystem destCRS  = dialog->crsObject();
+    if ( destCRS.isValid() )
+    {
+      QgsDatumTransformDialog::run( pclayer->crs(), destCRS, this, mMapCanvas );
+    }
+    exp.setCrs( destCRS );
+    exp.setFormat( dialog->format() );
+
+    if ( dialog->hasFilterExtent() )
+      exp.setFilterExtent( dialog->filterExtent() );
+
+    if ( ! dialog->layername().isEmpty() )
+      exp.setLayerName( dialog->layername() );
+
+    exp.setAttributes( dialog->selectedAttributes() );
+
+
+    vectorFilename = dialog->filename();
+    bool addToCanvas = dialog->addToCanvas();
+
+    QgsVectorLayer *result = exp.exportToMemoryLayer();
```

Comment:
This giant blocking operation on the main thread is not nice at all -- like the raster and vector export functions it should be moved to a background task instead.